Test.cs

```diff
@@ -12,7 +13,12 @@ public static long calculate(long number)
                 throw new Exception("Number must be non-negative.");
```

Comment:
Logical error in the factorial calculation. The recursive call should decrement the number, not increment it. Change `Calculate(number + 1)` to `Calculate(number - 1)`.